src/main/scala/mesosphere/marathon/core/volume/providers/DVDI.scala

```diff
@@ -0,0 +1,163 @@
+package mesosphere.marathon.core.volume.providers
+
+import com.wix.accord._
+import com.wix.accord.dsl._
+import com.wix.accord.Validator
+import mesosphere.marathon.core.volume._
+import mesosphere.marathon.state._
+import org.apache.mesos.Protos.{ ContainerInfo, Volume => MesosVolume, Environment }
+import scala.collection.JavaConverters._
+
+/**
+  * DVDIProvider (Docker Volume Driver Interface provider) handles persistent volumes allocated
+  * by a specific docker volume driver plugin. This works for both docker and mesos containerizers,
+  * albeit with some limitations:
+  *   - only a single volume driver per container is allowed when using the docker containerizer
+  *   - docker containerizer requires that referenced volumes be created prior to application launch
+  *   - mesos containerizer only supports volumes mounted in RW mode
+  */
+protected case object DVDIProvider extends InjectionHelper[PersistentVolume] with PersistentVolumeProvider {
+  import org.apache.mesos.Protos.Volume.Mode
+
+  override val name = Some("dvdi")
```

Comment:
we don't actually require the docker daemon for this feature (with the mesos containerizer) so I think "docker" is confusing. i didn't want a long name like "docker-volume-driver" because that gets very tedious when creating options prefixed with the provider name. i settled on `dvdi`. i like to think that i'm open to other options though :) is this something that we can mitigate with good documentation?